crypto/getenv.c

```diff
@@ -13,9 +13,62 @@
 
 #include <stdlib.h>
 #include "internal/cryptlib.h"
+#include "e_os.h"
 
 char *ossl_safe_getenv(const char *name)
 {
+#if defined(_WIN32) && defined(CP_UTF8) && !defined(_WIN32_WCE)
+    if (GetEnvironmentVariableW(L"OPENSSL_WIN32_UTF8", NULL, 0) != 0) {
+
+        char *val = NULL;
+        WCHAR *namew = NULL;
+        WCHAR *valw = NULL;
+        DWORD envlen = 0;
+        int rsize, fsize;
+
+        /* query for buffer len */
+        rsize = mbstowcs(NULL, name, 0) + 1;
+        namew = _malloca(rsize * sizeof(WCHAR));
+
+        if (NULL != namew) {
+            /* convert name to wide string */
+            fsize = mbstowcs(namew, name, rsize);
+            /* if conversion is ok, then determine value string size in wchars */
+            if (fsize > 0)
+                envlen = GetEnvironmentVariableW(namew, NULL, 0);
+        }
+
+        if (envlen > 0)
+            valw = _malloca(envlen * sizeof(WCHAR));
+
+        if (NULL != valw) {
+            if (GetEnvironmentVariableW(namew, valw, envlen) < envlen) {
+                /* determine value string size in utf-8 */
+                int sz = WideCharToMultiByte(CP_UTF8, 0, valw, -1,
+                                             NULL, 0, NULL, NULL);
+
+                if (sz != 0) {
+                    val = OPENSSL_malloc(sz);
```

Comment:
potentially val is NULL here if the malloc fails.. Does WideCharToMultiByte return 0 in this case? maybe checking for NULL is nicer?

Reply:
It's return 0, but I decide to rewrite code, thanks.

Reply:
You can probably see why this is not marked as trivial now :)

Reply:
It's done. I added one extra security check.